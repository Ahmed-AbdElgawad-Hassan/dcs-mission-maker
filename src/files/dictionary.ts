import { z } from "zod";

export const dictionarySchema = z.record(z.string(), z.string());

export type DictionaryProps = z.input<typeof dictionarySchema>;
export const dictionary = (props: DictionaryProps) =>
  dictionarySchema.parse(props);
