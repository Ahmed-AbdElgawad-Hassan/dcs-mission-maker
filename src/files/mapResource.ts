import { z } from "zod";

export const mapResourceSchema = z.object({});

export type MapResourceProps = z.input<typeof mapResourceSchema>;
export const mapResource = (props: MapResourceProps) =>
  mapResourceSchema.parse(props);
