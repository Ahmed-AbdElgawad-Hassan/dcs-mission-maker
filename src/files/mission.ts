import { z } from "zod";

const xPosSchema = z
  .number()
  .describe("X Vector Position, x is directed to the north");

const zPosSchema = z
  .number()
  .describe("Z Vector Position, z is directed to the east");

const dateSchema = z.object({
  Year: z
    .number()
    .min(1000)
    .max(3000)
    .describe("Mission Year as 4 digit year, i.e. 2016")
    .default(2016),
  Day: z
    .number()
    .min(1)
    .max(31)
    .describe("Mission Day of the Month between 1 and 31")
    .default(21),
  Month: z
    .number()
    .min(1)
    .max(12)
    .describe("Mission Month between 1 and 12")
    .default(6),
});

const startTimeSchema = z
  .number()
  .min(0)
  .max(86400)
  .describe("Mission Start time expressed in seconds since midnight")
  .default(28800);

const windAtSchema = z.object({
  speed: z.number().default(0),
  dir: z.number().default(0),
});

const windSchema = z.object({
  at8000: windAtSchema.default(windAtSchema.parse({})),
  atGround: windAtSchema.default(windAtSchema.parse({})),
  at2000: windAtSchema.default(windAtSchema.parse({})),
});

const visibilitySchema = z
  .object({ distance: z.number().default(80000) })
  .default({});

const haloSchema = z.object({ preset: z.string().default("auto") }).default({});

const fogSchema = z
  .object({
    thickness: z.number().default(0),
    visibility: z.number().default(0),
  })
  .default({});

const seasonSchema = z
  .object({ temperature: z.number().default(20) })
  .default({});

const cloudsSchema = z
  .object({
    thickness: z.number().default(200),
    density: z.number().default(0),
    preset: z.string().default("Preset2"),
    base: z.number().default(2500),
    iprecptns: z.number().default(0),
  })
  .default({});

const weatherSchema = z
  .object({
    atmosphere_type: z.number().default(0),
    wind: windSchema.default({}),
    enable_fog: z.boolean().default(false),
    visibility: visibilitySchema,
    halo: haloSchema,
    fog: fogSchema,
    season: seasonSchema,
    type_weather: z.number().default(0),
    qnh: z.number().default(760),
    cyclones: z.object({}).default({}),
    name: z.string().default("Winter, clean sky"),
    dust_density: z.number().default(0),
    modifiedTime: z.boolean().default(false),
    groundTurbulence: z.number().default(0),
    enable_dust: z.boolean().default(false),
    clouds: cloudsSchema,
  })
  .default({});

const sideResultSchema = z
  .object({
    conditions: z.object({}).default({}),
    actions: z.object({}).default({}),
    func: z.object({}).default({}),
  })
  .default({});

const resultSchema = z
  .object({
    total: z.number().default(0),
    offline: sideResultSchema,
    blue: sideResultSchema,
    red: sideResultSchema,
  })
  .default({});

const groundControlRoleSchema = z
  .object({
    blue: z.number().default(0),
    neutrals: z.number().default(0),
    red: z.number().default(0),
  })
  .default({});

const groundControlSchema = z
  .object({
    isPilotControlVehicles: z.boolean().default(false),
    passwords: z
      .object({
        artillery_commander: z.object({}).default({}),
        instructor: z.object({}).default({}),
        forward_observer: z.object({}).default({}),
        observer: z.object({}).default({}),
      })
      .default({}),
    roles: z
      .object({
        artillery_commander: groundControlRoleSchema,
        instructor: groundControlRoleSchema,
        forward_observer: groundControlRoleSchema,
        observer: groundControlRoleSchema,
      })
      .default({}),
  })
  .default({});

const coalitionsSchema = z
  .object({
    blue: z.array(z.number()).default([]),
    neutrals: z.array(z.number()).default([]),
    red: z.array(z.number()).default([]),
  })
  .default({});

const planeUnitSchema = z.object({
  alt: z.number().describe("Aircraft altitude in meters").default(2000),
  y: zPosSchema,
  x: xPosSchema,
  type: z.string(),
  onboard_num: z.string(),
  alt_type: z.string().default("BARO"),
  psi: z.number().default(0),
  livery_id: z.string().default("default"),
  skill: z.string().default("High"),
  callsign: z.object({
    _1: z.number(),
    _2: z.number(),
    _3: z.number(),
    name: z.string().nonempty(),
  }),
  name: z.string(),
  payload: z.object({
    pylons: z.object({}),
    fuel: z.string(),
    flare: z.number(),
    ammo_type: z.number().optional(),
    chaff: z.number(),
    gun: z.number(),
  }),
  speed: z.number().describe("Aircraft speed in meters per second"),
  heading: z.number().default(0),
  unitId: z.number(),
});

const planeRoutePointSchema = z.object({
  alt: z.number().default(2000),
  type: z.string().default("Turning Point"),
  action: z.string().default("Turning Point"),
  alt_type: z.string().default("BARO"),
  y: zPosSchema,
  x: xPosSchema,
  speed_locked: z.boolean(),
  formation_template: z.string(),
  speed: z.number(),
  ETA_locked: z.boolean(),
  task: z.object({
    id: z.string(),
    params: z.object({ tasks: z.object({}) }),
  }),
  ETA: z.number(),
});

const planeGroupSchema = z.object({
  frequency: z.number().default(124),
  modulation: z.number().default(0),
  groupId: z.number(),
  y: zPosSchema,
  x: xPosSchema,
  tasks: z.object({}).default({}),
  route: z.object({
    points: z.array(planeRoutePointSchema).default([]),
  }),
  hidden: z.boolean().default(false),
  units: z.array(planeUnitSchema).min(1).max(4),
  radioSet: z.boolean().default(false),
  name: z.string(),
  communication: z.boolean().default(true),
  start_time: z.number().default(0),
  task: z.string(),
  uncontrolled: z.boolean().default(false),
});

const coalitionCountrySchema = z.object({
  id: z.number(),
  name: z.string(),
  plane: z.object({
    group: z.array(planeGroupSchema).default([]),
  }),
});

const coalitionSchema = z.object({
  bullseye: z
    .object({ y: z.number().default(0), x: z.number().default(0) })
    .default({}),
  nav_points: z.object({}).default({}),
  name: z.enum(["blue", "neutrals", "red"]),
  country: z.array(coalitionCountrySchema).default([]),
});

export const missionSchema = z.object({
  requiredModules: z.object({}).default({}),
  date: dateSchema.default(dateSchema.parse({})),
  start_time: startTimeSchema,

  sortie: z.string(),

  descriptionText: z.string(),

  pictureFileNameB: z.object({}).default({}),
  descriptionBlueTask: z.string(),

  pictureFileNameN: z.object({}).default({}),
  descriptionNeutralsTask: z.string(),

  pictureFileNameR: z.object({}).default({}),
  descriptionRedTask: z.string(),

  theatre: z.string().describe("Theatre Name").default("Caucasus"),
  failures: z.object({}).default({}),
  currentKey: z.number().default(1),
  maxDictId: z.number(),

  weather: weatherSchema,
  result: resultSchema,
  groundControl: groundControlSchema,
  map: z
    .object({
      centerY: z.number().default(618000),
      zoom: z.number().default(1000000),
      centerX: z.number().default(-355000),
    })
    .default({}),
  coalitions: coalitionsSchema,
  triggers: z.object({ zones: z.object({}).default({}) }).default({}),
  version: z.number().default(1),
  goals: z.object({}).default({}),
  coalition: z
    .object({
      blue: coalitionSchema.default({ name: "blue" }),
      neutrals: coalitionSchema.default({ name: "neutrals" }),
      red: coalitionSchema.default({ name: "red" }),
    })
    .default({
      blue: { name: "blue" },
      red: { name: "red" },
      neutrals: { name: "neutrals" },
    }),
  trigrules: z.object({}).default({}),
  forcedOptions: z.object({}).default({}),
  trig: z
    .object({
      custom: z.object({}).default({}),
      customStartup: z.object({}).default({}),
      events: z.object({}).default({}),
      func: z.object({}).default({}),
      flag: z.object({}).default({}),
      conditions: z.object({}).default({}),
      actions: z.object({}).default({}),
      funcStartup: z.object({}).default({}),
    })
    .default({}),
});

export type MissionProps = z.input<typeof missionSchema>;
export const mission = (props: MissionProps) => missionSchema.parse(props);

export type PlaneProps = z.input<typeof planeUnitSchema>;
export const plane = (props: PlaneProps) => planeUnitSchema.parse(props);

export type PlaneGroupProps = z.input<typeof planeGroupSchema>;
export const planeGroup = (props: PlaneGroupProps) =>
  planeGroupSchema.parse(props);
