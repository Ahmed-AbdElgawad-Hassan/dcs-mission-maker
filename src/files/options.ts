import { z } from "zod";

export const optionsSchema = z.object({
  playerName: z.string().default("New callsign"),
  miscellaneous: z.object({}).default({}),
  difficulty: z.object({}).default({}),
  VR: z.object({}).default({}),
  graphics: z.object({}).default({}),
  plugins: z.object({}).default({}),
  format: z.number().default(1),
  sound: z.object({}).default({}),
  views: z.object({}).default({}),
});

export type OptionsProps = {
  playerName?: string;
};
export const options = ({ playerName }: OptionsProps) => {
  return optionsSchema.parse({ playerName });
};
