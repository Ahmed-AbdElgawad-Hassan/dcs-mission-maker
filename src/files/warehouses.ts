import { z } from "zod";

const airportFuelSchema = z.object({
  InitFuel: z.number().default(100),
});
const operatingLevelSchema = z.number().default(10);

const airportSchema = z.object({
  gasoline: airportFuelSchema,
  methanol_mixture: airportFuelSchema,
  diesel: airportFuelSchema,
  unlimitedMunitions: z.boolean().default(true),
  OperatingLevel_Air: operatingLevelSchema,
  speed: z.number().default(16.666666),
  size: z.number().default(100),
  periodicity: z.number().default(30),
  suppliers: z.object({}),
  coalition: z.string().default("NEUTRAL"),
  unlimitedAircrafts: z.boolean().default(true),
  OperatingLevel_Eqp: z.number().default(10),
  unlimitedFuel: z.boolean().default(true),
  aircrafts: z.object({}),
  weapons: z.object({}),
  OperatingLevel_Fuel: operatingLevelSchema,
  jet_fuel: airportFuelSchema,
});

const warehousesSchema = z.object({
  airports: z.record(z.number(), airportSchema).default({}),
  warehouses: z.record(z.number(), z.object({})).default({}),
});

export type WarehousesProps = {};
export const warehouses = (props: WarehousesProps) => {
  return warehousesSchema.parse({});
};
