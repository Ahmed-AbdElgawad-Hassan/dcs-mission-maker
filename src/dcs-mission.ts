import JSZip from "jszip";
import { js2Lua } from "./js-2-lua";
import { options } from "./files/options";
import { mapResource } from "./files/mapResource";
import { warehouses } from "./files/warehouses";
import { mission } from "./files/mission";
import { dictionary } from "./files/dictionary";
import { Dictionary } from "./dictionary";

export type DcsMissionProps = {
  theatre: string;
};

export class DcsMission {
  public readonly dictionary: Dictionary;

  constructor(public readonly props: DcsMissionProps) {
    this.dictionary = new Dictionary();
  }

  getFiles() {
    return [
      ["options", `options = ${js2Lua(options({}))}`],
      ["l10n/DEFAULT/mapResource", `mapResource = ${js2Lua(mapResource({}))}`],
      ["warehouses", `warehouses = ${js2Lua(warehouses({}))}`],
      ["theatre", this.props.theatre],
      [
        "mission",
        `mission = ${js2Lua(
          mission({
            maxDictId: this.dictionary.maxDictId,
            theatre: this.props.theatre,
            descriptionText: this.dictionary.addEntry("descriptionText", ""),
            descriptionRedTask: this.dictionary.addEntry(
              "descriptionRedTask",
              "",
            ),
            descriptionBlueTask: this.dictionary.addEntry(
              "descriptionBlueTask",
              "",
            ),
            descriptionNeutralsTask: this.dictionary.addEntry(
              "descriptionNeutralsTask",
              "",
            ),
            sortie: this.dictionary.addEntry("sortie", ""),
          }),
        )}`,
      ],
      [
        "l10n/DEFAULT/dictionary",
        `dictionary = ${js2Lua(dictionary(this.dictionary.entries))}`,
      ],
    ];
  }

  build() {
    const zip = new JSZip();
    this.getFiles().forEach(([path, data]) => zip.file(path, data));
    return zip.generateAsync({ compression: "DEFLATE", type: "arraybuffer" });
  }
}
