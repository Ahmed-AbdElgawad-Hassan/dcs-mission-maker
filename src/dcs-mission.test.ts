import { DcsMission } from "./dcs-mission";
import { plane, planeGroup } from "./files/mission";

describe("given a simple empty mission", () => {
  it("should build a mission file", () => {
    const dcsMission = new DcsMission({
      theatre: "Caucasus",
      descriptionNeutralsTask: "",
      descriptionRedTask: "",
      descriptionBlueTask: "",
      sortie: "",
      descriptionText: "",
    });
    expect(dcsMission.getFiles()).toMatchSnapshot();
  });
});

describe("given a simple mission with 1 plane group", () => {
  it("should build a mission file", () => {
    const dcsMission = new DcsMission({
      theatre: "Caucasus",
      descriptionNeutralsTask: "",
      descriptionRedTask: "",
      descriptionBlueTask: "",
      sortie: "",
      descriptionText: "",
      coalitions: { blue: [80] },
      coalition: {
        blue: {
          name: "blue",
          country: [
            {
              id: 80,
              name: "CJTF Blue",
              plane: {
                group: [
                  planeGroup({
                    task: "CAS",
                    groupId: 1,
                    name: "Aerial-1",
                    x: -297153.83834764,
                    y: 632449.86255341,
                    route: {
                      points: [
                        {
                          alt: 2000,
                          type: "Turning Point",
                          action: "Turning Point",
                          alt_type: "BARO",
                          y: 632449.86255341,
                          x: -297153.83834764,
                          speed_locked: true,
                          formation_template: "",
                          speed: 154.16666666667,
                          ETA_locked: true,
                          task: {
                            id: "ComboTask",
                            params: {
                              tasks: {},
                            },
                          },
                          ETA: 0,
                        },
                      ],
                    },
                    units: [
                      plane({
                        type: "A-10A",
                        onboard_num: "010",
                        callsign: { _1: 1, _2: 1, _3: 1, name: "Enfield11" },
                        y: 635588.6,
                        x: -292598.45714286,
                        name: "Aerial-1-1",
                        payload: {
                          pylons: {},
                          fuel: "5029",
                          flare: 120,
                          ammo_type: 1,
                          chaff: 240,
                          gun: 100,
                        },
                        speed: 123.33333333333,
                        AddPropAircraft: {},
                        unitId: 1,
                      }),
                    ],
                  }),
                ],
              },
            },
          ],
        },
      },
    });
    expect(dcsMission.getFiles()).toMatchSnapshot();
  });
});
