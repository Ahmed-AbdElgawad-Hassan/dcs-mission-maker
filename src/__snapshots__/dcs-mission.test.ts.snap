// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given a simple empty mission should build a mission file 1`] = `
[
  [
    "options",
    "options = 
{
	["playerName"]="New callsign",
	["miscellaneous"]=
	{
	},
	["difficulty"]=
	{
	},
	["VR"]=
	{
	},
	["graphics"]=
	{
	},
	["plugins"]=
	{
	},
	["format"]=1,
	["sound"]=
	{
	},
	["views"]=
	{
	}
}",
  ],
  [
    "l10n/DEFAULT/mapResource",
    "mapResource = 
{
}",
  ],
  [
    "warehouses",
    "warehouses = 
{
	["airports"]=
	{
	},
	["warehouses"]=
	{
	}
}",
  ],
  [
    "theatre",
    "Caucasus",
  ],
  [
    "mission",
    "mission = 
{
	["requiredModules"]=
	{
	},
	["date"]=
	{
		["Year"]=2016,
		["Day"]=21,
		["Month"]=6
	},
	["start_time"]=28800,
	["sortie"]="DictKey_sortie_4",
	["descriptionText"]="DictKey_descriptionText_0",
	["pictureFileNameB"]=
	{
	},
	["descriptionBlueTask"]="DictKey_descriptionBlueTask_1",
	["pictureFileNameN"]=
	{
	},
	["descriptionNeutralsTask"]="DictKey_descriptionNeutralsTask_3",
	["pictureFileNameR"]=
	{
	},
	["descriptionRedTask"]="DictKey_descriptionRedTask_2",
	["theatre"]="Caucasus",
	["failures"]=
	{
	},
	["currentKey"]=1,
	["maxDictId"]=5,
	["weather"]=
	{
		["atmosphere_type"]=0,
		["wind"]=
		{
			["at8000"]=
			{
				["speed"]=0,
				["dir"]=0
			},
			["atGround"]=
			{
				["speed"]=0,
				["dir"]=0
			},
			["at2000"]=
			{
				["speed"]=0,
				["dir"]=0
			}
		},
		["enable_fog"]=false,
		["visibility"]=
		{
			["distance"]=80000
		},
		["halo"]=
		{
			["preset"]="auto"
		},
		["fog"]=
		{
			["thickness"]=0,
			["visibility"]=0
		},
		["season"]=
		{
			["temperature"]=20
		},
		["type_weather"]=0,
		["qnh"]=760,
		["cyclones"]=
		{
		},
		["name"]="Winter, clean sky",
		["dust_density"]=0,
		["modifiedTime"]=false,
		["groundTurbulence"]=0,
		["enable_dust"]=false,
		["clouds"]=
		{
			["thickness"]=200,
			["density"]=0,
			["preset"]="Preset2",
			["base"]=2500,
			["iprecptns"]=0
		}
	},
	["result"]=
	{
		["total"]=0,
		["offline"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		},
		["blue"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		},
		["red"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		}
	},
	["groundControl"]=
	{
		["isPilotControlVehicles"]=false,
		["passwords"]=
		{
			["artillery_commander"]=
			{
			},
			["instructor"]=
			{
			},
			["forward_observer"]=
			{
			},
			["observer"]=
			{
			}
		},
		["roles"]=
		{
			["artillery_commander"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["instructor"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["forward_observer"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["observer"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			}
		}
	},
	["map"]=
	{
		["centerY"]=618000,
		["zoom"]=1000000,
		["centerX"]=-355000
	},
	["coalitions"]=
	{
		["blue"]=
		{
		},
		["neutrals"]=
		{
		},
		["red"]=
		{
		}
	},
	["triggers"]=
	{
		["zones"]=
		{
		}
	},
	["version"]=1,
	["goals"]=
	{
	},
	["coalition"]=
	{
		["blue"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="blue",
			["country"]=
			{
			}
		},
		["neutrals"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="neutrals",
			["country"]=
			{
			}
		},
		["red"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="red",
			["country"]=
			{
			}
		}
	},
	["trigrules"]=
	{
	},
	["forcedOptions"]=
	{
	},
	["trig"]=
	{
		["custom"]=
		{
		},
		["customStartup"]=
		{
		},
		["events"]=
		{
		},
		["func"]=
		{
		},
		["flag"]=
		{
		},
		["conditions"]=
		{
		},
		["actions"]=
		{
		},
		["funcStartup"]=
		{
		}
	}
}",
  ],
  [
    "l10n/DEFAULT/dictionary",
    "dictionary = 
{
	["DictKey_descriptionText_0"]="",
	["DictKey_descriptionBlueTask_1"]="",
	["DictKey_descriptionRedTask_2"]="",
	["DictKey_descriptionNeutralsTask_3"]="",
	["DictKey_sortie_4"]=""
}",
  ],
]
`;

exports[`given a simple mission with 1 plane group should build a mission file 1`] = `
[
  [
    "options",
    "options = 
{
	["playerName"]="New callsign",
	["miscellaneous"]=
	{
	},
	["difficulty"]=
	{
	},
	["VR"]=
	{
	},
	["graphics"]=
	{
	},
	["plugins"]=
	{
	},
	["format"]=1,
	["sound"]=
	{
	},
	["views"]=
	{
	}
}",
  ],
  [
    "l10n/DEFAULT/mapResource",
    "mapResource = 
{
}",
  ],
  [
    "warehouses",
    "warehouses = 
{
	["airports"]=
	{
	},
	["warehouses"]=
	{
	}
}",
  ],
  [
    "theatre",
    "Caucasus",
  ],
  [
    "mission",
    "mission = 
{
	["requiredModules"]=
	{
	},
	["date"]=
	{
		["Year"]=2016,
		["Day"]=21,
		["Month"]=6
	},
	["start_time"]=28800,
	["sortie"]="DictKey_sortie_4",
	["descriptionText"]="DictKey_descriptionText_0",
	["pictureFileNameB"]=
	{
	},
	["descriptionBlueTask"]="DictKey_descriptionBlueTask_1",
	["pictureFileNameN"]=
	{
	},
	["descriptionNeutralsTask"]="DictKey_descriptionNeutralsTask_3",
	["pictureFileNameR"]=
	{
	},
	["descriptionRedTask"]="DictKey_descriptionRedTask_2",
	["theatre"]="Caucasus",
	["failures"]=
	{
	},
	["currentKey"]=1,
	["maxDictId"]=5,
	["weather"]=
	{
		["atmosphere_type"]=0,
		["wind"]=
		{
			["at8000"]=
			{
				["speed"]=0,
				["dir"]=0
			},
			["atGround"]=
			{
				["speed"]=0,
				["dir"]=0
			},
			["at2000"]=
			{
				["speed"]=0,
				["dir"]=0
			}
		},
		["enable_fog"]=false,
		["visibility"]=
		{
			["distance"]=80000
		},
		["halo"]=
		{
			["preset"]="auto"
		},
		["fog"]=
		{
			["thickness"]=0,
			["visibility"]=0
		},
		["season"]=
		{
			["temperature"]=20
		},
		["type_weather"]=0,
		["qnh"]=760,
		["cyclones"]=
		{
		},
		["name"]="Winter, clean sky",
		["dust_density"]=0,
		["modifiedTime"]=false,
		["groundTurbulence"]=0,
		["enable_dust"]=false,
		["clouds"]=
		{
			["thickness"]=200,
			["density"]=0,
			["preset"]="Preset2",
			["base"]=2500,
			["iprecptns"]=0
		}
	},
	["result"]=
	{
		["total"]=0,
		["offline"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		},
		["blue"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		},
		["red"]=
		{
			["conditions"]=
			{
			},
			["actions"]=
			{
			},
			["func"]=
			{
			}
		}
	},
	["groundControl"]=
	{
		["isPilotControlVehicles"]=false,
		["passwords"]=
		{
			["artillery_commander"]=
			{
			},
			["instructor"]=
			{
			},
			["forward_observer"]=
			{
			},
			["observer"]=
			{
			}
		},
		["roles"]=
		{
			["artillery_commander"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["instructor"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["forward_observer"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			},
			["observer"]=
			{
				["blue"]=0,
				["neutrals"]=0,
				["red"]=0
			}
		}
	},
	["map"]=
	{
		["centerY"]=618000,
		["zoom"]=1000000,
		["centerX"]=-355000
	},
	["coalitions"]=
	{
		["blue"]=
		{
			[1]=80
		},
		["neutrals"]=
		{
		},
		["red"]=
		{
		}
	},
	["triggers"]=
	{
		["zones"]=
		{
		}
	},
	["version"]=1,
	["goals"]=
	{
	},
	["coalition"]=
	{
		["blue"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="blue",
			["country"]=
			{
				[1]=
				{
					["id"]=80,
					["name"]="CJTF Blue",
					["plane"]=
					{
						["group"]=
						{
							[1]=
							{
								["frequency"]=124,
								["modulation"]=0,
								["groupId"]=1,
								["y"]=632449.86255341,
								["x"]=-297153.83834764,
								["tasks"]=
								{
								},
								["route"]=
								{
									["points"]=
									{
										[1]=
										{
											["alt"]=2000,
											["type"]="Turning Point",
											["action"]="Turning Point",
											["alt_type"]="BARO",
											["y"]=632449.86255341,
											["x"]=-297153.83834764,
											["speed_locked"]=true,
											["formation_template"]="",
											["speed"]=154.16666666667,
											["ETA_locked"]=true,
											["task"]=
											{
												["id"]="ComboTask",
												["params"]=
												{
													["tasks"]=
													{
													}
												}
											},
											["ETA"]=0
										}
									}
								},
								["hidden"]=false,
								["units"]=
								{
									[1]=
									{
										["alt"]=2000,
										["y"]=635588.6,
										["x"]=-292598.45714286,
										["type"]="A-10A",
										["onboard_num"]="010",
										["alt_type"]="BARO",
										["psi"]=0,
										["livery_id"]="default",
										["skill"]="High",
										["callsign"]=
										{
											[1]=1,
											[2]=1,
											[3]=1,
											["name"]="Enfield11"
										},
										["name"]="Aerial-1-1",
										["payload"]=
										{
											["pylons"]=
											{
											},
											["fuel"]="5029",
											["flare"]=120,
											["ammo_type"]=1,
											["chaff"]=240,
											["gun"]=100
										},
										["speed"]=123.33333333333,
										["heading"]=0,
										["unitId"]=1
									}
								},
								["radioSet"]=false,
								["name"]="Aerial-1",
								["communication"]=true,
								["start_time"]=0,
								["task"]="CAS",
								["uncontrolled"]=false
							}
						}
					}
				}
			}
		},
		["neutrals"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="neutrals",
			["country"]=
			{
			}
		},
		["red"]=
		{
			["bullseye"]=
			{
				["y"]=0,
				["x"]=0
			},
			["nav_points"]=
			{
			},
			["name"]="red",
			["country"]=
			{
			}
		}
	},
	["trigrules"]=
	{
	},
	["forcedOptions"]=
	{
	},
	["trig"]=
	{
		["custom"]=
		{
		},
		["customStartup"]=
		{
		},
		["events"]=
		{
		},
		["func"]=
		{
		},
		["flag"]=
		{
		},
		["conditions"]=
		{
		},
		["actions"]=
		{
		},
		["funcStartup"]=
		{
		}
	}
}",
  ],
  [
    "l10n/DEFAULT/dictionary",
    "dictionary = 
{
	["DictKey_descriptionText_0"]="",
	["DictKey_descriptionBlueTask_1"]="",
	["DictKey_descriptionRedTask_2"]="",
	["DictKey_descriptionNeutralsTask_3"]="",
	["DictKey_sortie_4"]=""
}",
  ],
]
`;
